crypto/keys/internal/ecdsa/privkey.go

```diff
@@ -0,0 +1,69 @@
+package ecdsa
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/sha256"
+	"fmt"
+	"math/big"
+)
+
+// GenPrivKey generates a new secp256r1 private key. It uses operating system randomness.
+func GenPrivKey(curve elliptic.Curve) (PrivKey, error) {
+	key, err := ecdsa.GenerateKey(curve, rand.Reader)
+	if err != nil {
+		return PrivKey{}, err
+	}
+	return PrivKey{*key}, nil
+}
+
+type PrivKey struct {
+	ecdsa.PrivateKey
+}
+
+// PubKey implements Cosmos-SDK PrivKey interface.
+func (sk *PrivKey) PubKey() PubKey {
+	return PubKey{sk.PublicKey, nil}
+}
+
+// Bytes serialize the private key with first byte being the curve type.
+func (sk *PrivKey) Bytes() []byte {
+	if sk == nil {
+		return nil
+	}
+	fieldSize := (sk.Curve.Params().BitSize + 7) / 8
+	bz := make([]byte, fieldSize)
+	sk.D.FillBytes(bz)
+	return bz
+}
+
+// Sign hashes and signs the message usign ECDSA. Implements sdk.PrivKey interface.
+func (sk *PrivKey) Sign(msg []byte) ([]byte, error) {
+	digest := sha256.Sum256(msg)
```

Comment:
are you sure this is needed, the stdlib doesn't do it under the hood already? (genuine question, i have no idea)

Reply:
Yes. `ed25519` creates a hash (it's a part of the specification). The `ECDSA` doesn't specify it. The STD lib only guides in the documentation that the parameter should be a digest - so we must hash before:
```
// stdlib:
Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
```

Reply:
Sounds good, thanks!